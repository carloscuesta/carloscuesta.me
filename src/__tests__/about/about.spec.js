import renderer from 'react-test-renderer'

import About from 'src/pages/about'

jest.mock('next-seo', () => ({ NextSeo: 'NextSeo', SocialProfileJsonLd: 'SocialProfileJsonLd' }))

/*
  I'm mocking the components since those are unit tested.
  So there's no point to test the rendering of them twice.
  It's more easier to manage little snapshots than big ones.
*/

jest.mock('src/components/pages/about/Contact', () => 'Contact')
jest.mock('src/components/pages/about/Education', () => 'Education')
jest.mock('src/components/pages/about/Experience', () => 'Experience')
jest.mock('src/components/pages/about/ItemTimestamp', () => 'ItemTimestamp')
jest.mock('src/components/pages/about/ItemTitle', () => 'ItemTitle')
jest.mock('src/components/pages/about/WhoAmI', () => 'WhoAmI')

describe('about', () => {
  describe('page', () => {
    it('should match about page', () => {
      const wrapper = renderer.create(<About />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
