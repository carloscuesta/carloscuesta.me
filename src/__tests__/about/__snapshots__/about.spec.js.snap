// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about page should match about page 1`] = `
Array [
  <NextSeo
    canonical="https://carloscuesta.me/about"
    title="Carlos Cuesta â€“ About me"
  />,
  <SocialProfileJsonLd
    name="Carlos Cuesta"
    sameAs={
      Array [
        "http://instagram.com/crloscuesta",
        "https://github.com/carloscuesta",
        "https://twitter.com/crloscuesta",
        "https://www.linkedin.com/in/crloscuesta",
      ]
    }
    type="Person"
    url="https://carloscuesta.me"
  />,
  <main>
    <WhoAmI />
    <Experience />
    <Education />
    <Contact />
  </main>,
]
`;
